eZ/Publish/Core/FieldType/Image/ImageThumbnailStrategy.php

```diff
@@ -43,12 +53,46 @@ public function getFieldTypeIdentifier(): string
 
     public function getThumbnail(Field $field, ?APIVersionInfo $versionInfo = null): ?Thumbnail
     {
-        /** @var \eZ\Publish\SPI\Variation\Values\ImageVariation $variation */
-        $variation = $this->variationHandler->getVariation(
-            $field,
-            $versionInfo ?? new VersionInfo(),
-            $this->variationName
-        );
+        try {
+            /** @var \eZ\Publish\SPI\Variation\Values\ImageVariation $variation */
+            $variation = $this->variationHandler->getVariation(
+                $field,
+                $versionInfo ?? new VersionInfo(),
+                $this->variationName
+            );
+        } catch (SourceImageNotFoundException $e) {
+            $contentDetails = isset($versionInfo->contentInfo) && isset($versionInfo->versionNo)
+                ? sprintf(' Content: %d, Version No: %d', $versionInfo->contentInfo->id, $versionInfo->versionNo)
+                : '';
```

Comment:
It's not necessary to check properties here. Also we tend not to use magic properties if there's an alternative. What is simply enough:
```suggestion
            $contentDetails = null !== $versionInfo
                ? sprintf(' Content: %d, Version No: %d', $versionInfo->getContentInfo()->id, $versionInfo->versionNo)
                : '';
```